nolearn/lasagne/base.py

```diff
@@ -316,7 +324,9 @@ def train_loop(self, X, y):
             min([row['train_loss'] for row in self.train_history_]) if
             self.train_history_ else np.inf
             )
-        first_iteration = True
+        for func in on_training_started:
+            func(self)
```

Comment:
I would say that for consistency, we should make the interface the same as for the other handlers: `func(self, self.train_history_)`, where `train_history`, as you know, might have values already.